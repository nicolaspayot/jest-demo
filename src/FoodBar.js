import { api } from './api.js';

export default {
  template: /*html*/`
    <div class="main">
      <h1>Welcome to Vue food bar!</h1>
      <span class="is-large">🍔</span> <button id="burger" @click="add('🍔')">Add 1</button>
      <span class="is-large">🥤</span> <button id="drink" @click="add('🥤')">Add 1</button>
      <span class="is-large">🍟</span> <button id="fries" @click="add('🍟')">Add 1</button>
      <span class="is-large">🍦</span> <button id="ice-cream" @click="add('🍦')">Add 1</button>
      <div id="menu" v-if="menu.length > 0">
        <h2>Here's what your ordered:</h2>
        <div class="is-large">
          <span>👉</span>
          <span>{{ menu.join('') }}</span>
        </div>
        <div class="checkout"><button @click="checkout">Checkout!</button></div>
      </div>
    </div>
  `,
  data () {
    return {
      menu: []
    }
  },
  methods: {
    add (item) {
      this.menu.push(item);
    },
    checkout () {
      return api.checkout().then(() => {
        this.$router.push('/confirm');
      });
    }
  }
};
