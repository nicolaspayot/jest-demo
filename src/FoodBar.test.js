jest.mock('./api');
import { shallow } from '@vue/test-utils';
import FoodBar from './FoodBar';

describe('FoodBar component', () => {
  let wrapper;

  beforeEach(() => {
    const $router = { push: jest.fn() };
    wrapper = shallow(FoodBar, {
      mocks: { $router }
    });
  });

  it('should display orders', () => {
    wrapper.find('#burger').trigger('click');
    wrapper.find('#burger').trigger('click');
    wrapper.find('#burger').trigger('click');
    wrapper.find('#drink').trigger('click');
    wrapper.find('#fries').trigger('click');
    wrapper.find('#ice-cream').trigger('click');

    expect(wrapper.find('#menu').html()).toBe("<div id=\"menu\"><h2>Here's what your ordered:</h2> <div class=\"is-large\"><span>👉</span> <span>🍔🍔🍔🥤🍟🍦</span></div> <div class=\"checkout\"><button>Checkout!</button></div></div>");
  });

  it('should match snapshot', () => {
    wrapper.find('#burger').trigger('click');
    wrapper.find('#fries').trigger('click');
    wrapper.find('#drink').trigger('click');
    wrapper.find('#ice-cream').trigger('click');

    const renderer = require('vue-server-renderer').createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot();
    });
  });

  it('should confirm order', async () => {
    wrapper.find('#burger').trigger('click');
    wrapper.find('#fries').trigger('click');

    await wrapper.vm.checkout();
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith('/confirm');
  });
});
