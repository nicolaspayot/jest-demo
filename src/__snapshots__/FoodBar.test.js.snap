// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodBar component should match snapshot 1`] = `
<div class="main">
    <h1>Welcome to Vue food bar!</h1> <span class="is-large">🍔</span>
    <button id="burger">Add 1</button> <span class="is-large">🥤</span>
    <button id="drink">Add 1</button> <span class="is-large">🍟</span>
    <button id="fries">Add 1</button> <span class="is-large">🍦</span>
    <button id="ice-cream">Add 1</button>
    <div id="menu">
        <h2>Here's what your ordered:</h2>
        <div class="is-large"><span>👉</span> <span>🍔🍟🥤🍦</span></div>
        <div class="checkout">
            <button>Checkout!</button>
        </div>
    </div>
</div>
`;
